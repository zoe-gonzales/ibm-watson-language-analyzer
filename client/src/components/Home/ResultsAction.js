import { useState, useEffect } from 'react';

const ResultsAction = apiResults => {
    const [results, setResults] = useState([]);
    const [type, setType] = useState('');
    const [text, setText] = useState('');

    useEffect(() => {
        if (apiResults !== {}) {
            setResults(apiResults.results);
            setType(apiResults.type);
            setText(apiResults.text);
        }
    }, []);

    return {
        results,
        type,
        text
    }
}

export default ResultsAction;