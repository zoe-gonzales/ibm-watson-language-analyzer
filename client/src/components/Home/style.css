@media screen and (max-width: 1024px) {    
    .label {
        font-weight: bold;
    }
}

@media screen and (max-width: 360px) {
    .txt {
        font-size: 1rem;
    }
}