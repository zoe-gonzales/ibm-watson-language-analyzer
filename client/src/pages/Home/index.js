import React, { useState, useEffect } from 'react';
import Grid from '../../components/ResultsGrid';
import Title from '../../components/Title';
import Directions from '../../components/Directions';
import Form from '../../components/Form';

const Home = ({ apiResults }) => {
    const [results, setResults] = useState([]);
    const [type, setType] = useState('');
    const [text, setText] = useState('');

    useEffect(() => {
        if (apiResults !== {}) {
            setResults(apiResults.results);
            setType(apiResults.type);
            setText(apiResults.text);
        }
    }, []);

    const dirs = [
        {
            id: 1,
            text: 'Enter text to analyze'
        },
        {
            id: 2,
            text: 'Select method'
        },
        {
            id: 3,
            text: 'Get results!'
        }
    ]

    return (
        <div className='container'>
        {results.length > 0 ? (
            <Grid type={type} text={text} results={results} />
        ) : (
            <div>
                <Title>Language Analyzer</Title>
                <Directions dirs={dirs} />
                <Form />
            </div>
        )}
        </div>
    )
}

export default Home;