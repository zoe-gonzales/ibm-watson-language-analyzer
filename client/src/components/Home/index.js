import React from 'react';
import Grid from '../../components/ResultsGrid';
import Title from '../../components/Title';
import Directions from '../../components/Directions';
import Form from '../../components/Form';
import ResultsAction from './ResultsAction';
import './style.css';

const Home = ({ apiResults }) => {
    const { results, type, text } = ResultsAction(apiResults);
    const dirs = [
        {
            id: 1,
            text: 'Enter text to analyze'
        },
        {
            id: 2,
            text: 'Select method'
        },
        {
            id: 3,
            text: 'Get results!'
        }
    ];

    return (
        <div className='container'>
        {results ? (
            <Grid type={type} text={text} results={results} />
        ) : (
            <div>
                <Title>Language Analyzer</Title>
                <Directions dirs={dirs} />
                <Form />
            </div>
        )}
        </div>
    )
}

export default Home;