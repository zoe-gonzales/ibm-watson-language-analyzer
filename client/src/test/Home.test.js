import React from 'react';
import Home from '../pages/Home';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import sinon from 'sinon';

describe('Home page', () => {
  afterEach(() => sinon.restore());

  it('renders correctly', () => {
    const component = renderer
      .create(<Home apiResults={{}} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('calls Home component once with no data', () => {
    const spy = sinon.spy(Home);
    expect(spy.calledOnce);
    expect(spy.calledWith({}));
  });

  it('calls Home component with data', () => {
    const results = { data: 'some data' };
    const wrapper = mount(<Home apiResults={results} />);
    expect(wrapper.props().apiResults).to.equal(results);
  });
});