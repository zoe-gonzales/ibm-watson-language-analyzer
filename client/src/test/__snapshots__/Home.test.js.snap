// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<div
  className="container"
>
  <div>
    <div
      className="row"
    >
      <div
        className="col s4"
      >
        Text
      </div>
      <div
        className="col s8"
      >
        
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col s4"
      >
        
        s
      </div>
      <div
        className="col s4"
      >
        
      </div>
      <div
        className="col s4"
      >
        Relevance
      </div>
    </div>
  </div>
</div>
`;
